ckan/tests/logic/test_validators.py

```diff
@@ -294,6 +294,25 @@ def call_validator(*args, **kwargs):
         call_validator(valid_value)
 
 
+def test_strip_value_with_valid_value():
+    valid_values = [
+        " test@example.com",
+        "  test@example.com",
+        "test@example.com ",
+        "test@example.com  ",
+        " test@example.com ",
+        "  test@example.com  ",
+    ]
+
+    for valid_value in valid_values:
+
+        @returns_arg
+        def call_validator(*args):
+            return validators.strip_value(*args)
+
+        call_validator(valid_value)
```

Comment:
You are getting a failure because we can't use `@returns_arg`. As its docstring says, it's a decorator used to check that the returned value is the same as the provided one, and in your case you are changing the value (trimming it). Just call the validator directly and check it removes the whitespace:

```suggestion
             assert validators.strip_value(valid_value) == '"test@example.com"
```